type: enable zero value for BackendStore

fatih opened the pull request:
Problem: There is no way one can identify if a BackendStore is
uninitialized (a.k.a has zero value) or initialized. It always will
return the type `Btrfs`. Also the `String()` method never will return the
string `<INVALID>`

Solution: Increase the enum just one by one. This will make
uninitialized values of BackendStore as 0. One can then check easily
with:

```
store := BackendStore

if store == 0 {
    // uninitialized
}
```

Also the `String()` methods is going to return the `<INVALID>` string.

caglar10ur commented:
Maybe alternatively we can make Directory backendstore type the first one as it's the default in LXC world (like below) and assume that uninitialized BackendStore means Directory. Do you think that make sense?

```
diff --git a/type.go b/type.go
index f63b1d3..2329a65 100644
--- a/type.go
+++ b/type.go
@@ -30,10 +30,10 @@ const (
 type BackendStore int

 const (
-       // Btrfs backendstore type
-       Btrfs BackendStore = iota
        // Directory backendstore type
-       Directory
+       Directory = iota
+       // Btrfs backendstore type
+       Btrfs BackendStore
        // LVM backendstore type
        LVM
        // ZFS backendstore type
```

By the way starting with LXC 1.0.4 we also started to export bdev_specs (https://github.com/lxc/lxc/commit/35d0ebf102ff1ec11340593f214fa1b3113b991d) as well and I'm working on some changes to introduce that support as some Backends depend on that.

fatih commented:
That would be nice. But my only concern is to find if the value is initialized intentionally or not. In a RPC system I'm going to receive the method arguments in a struct like : 

``` go
type Args struct {
    Template string
    Backend  lxc.BackendStore
}
```

Now I can easily test and give a feedback back:

``` go
if args.Backend == 0 {
    return errors.New("create: backend field is required.")
}
```

It comes down how this API is being used and acting. I'm usually against implicit behaviors. Or any side effect that I'm not known. In my opinion we should design the API in an explicit way. It should be predictable and any misused values should be return as an error (like the `<INVALID>` string). This would allow us to create robust applications on top of it.

fatih commented:
Btw the `State` type also has the same behavior:

https://github.com/lxc/go-lxc/blob/master/type.go#L79

caglar10ur commented:
OK I think it makes sense. Do you mind fixing the State as well?

fatih committed:
type: zero values for State and BackendStore types

Problem: There is no way one can identify if a type is uninitialized
(a.k.a has zero value) or initalized for State or BackendStore types. It
always will return the type STOPPED (or Btrfs). Also the String() method
never will return the string "<INVALID>".

Solution: Increase the enum just by one. This will make uninitialized
values of BackendStore and State as 0. One can then check easily with:

	store := BackendStore
	if store == 0 {
 		return errors.New("backend is not initalized.")
	}

Also the String() method is going to return the "<INVALID>" string.

fatih commented:
Done, pushed it as well (and rebased). Added a separate test for it too.

caglar10ur commented:
Merged, thanks!
